modules/dxp/apps/search-experiences/search-experiences-web/src/main/resources/META-INF/resources/semantic_search/js/configuration/ModelAutocomplete.js

```diff
@@ -19,43 +20,107 @@ import React, {useState} from 'react';
  * will query for applicable models.
  * This can be found on: System Settings > Search Experiences > Semantic Search
  */
-function ModelAutocomplete({label, name, onChange, required, value = ''}) {
-	const [networkStatus, setNetworkStatus] = useState(4);
+function ModelAutocomplete({
+	label,
+	name,
+	onBlur,
+	onChange,
+	required,
+	value = '',
+}) {
+	const [autocompleteSearchValue, setAutocompleteSearchValue] = useState(
+		value
+	);
+	const [networkState, setNetworkState] = useState(() => ({
+		error: false,
+		loading: false,
+		networkStatus: 4,
+	}));
+	const [showDropDown, setShowDropDown] = useState(false);
 
 	const {resource} = useResource({
 		fetchPolicy: 'cache-first',
 		link: `${window.location.origin}${Liferay.ThemeDisplay.getPathContext()}
 		/o/search-experiences-rest/v1.0/sentence-transformer/ml-models`,
-		onNetworkStatusChange: setNetworkStatus,
+		onNetworkStatusChange: (status) => {
+			setNetworkState({
+				error: status === 5,
+				loading: status > 1 && status < 4,
+				networkStatus: status,
+			});
+		},
 		variables: {
 			limit: 20,
 			pipeline_tag: 'feature-extraction',
-			query: value,
+			query: autocompleteSearchValue,
 		},
 	});
 
+	const _handleBlur = () => {
+		if (!autocompleteSearchValue) {
+			onChange('');
+		}
+		else if (value !== autocompleteSearchValue) {
+			setAutocompleteSearchValue(value);
+		}
+
+		onBlur();
+	};
+
+	const _handleFocus = () => {
+		setShowDropDown(true);
+	};
+
+	const _handleInputChange = (event) => {
+		setShowDropDown(true);
+		setAutocompleteSearchValue(event.target.value);
+	};
+
+	const _handleItemChange = (item) => {
+		setAutocompleteSearchValue(item);
+		onChange(item);
+		setShowDropDown(false);
```

Comment:
If lines are grouped together, they should be sorted. If they shouldn't be, use a blank line in between.

```
setAutocompleteSearchValue(event.target.value);
setShowDropDown(true);
```

Reply:
Added!